### ISSUE TITLE
tf.nn.softmax -- the "name" argument is passed to the internal flattened tensor

### ISSUE THREAD

**Issue by ilya-edrenkin** [tensorflow/tensorflow#8667]
Hi, if I use tf.nn.softmax with the `name` argument, I expect the output tensor to be given this name.
However, the output tensor is always a Reshape op named `Reshape_...`, and the passed name refers to an internal (flattened) tensor.

Steps to reproduce (TF v1.0.1):
```
In [1]: import tensorflow as tf
In [2]: a = tf.nn.softmax(tf.ones([2, 8, 20]), name="softmax_custom_name")
In [3]: print a
Tensor("Reshape_1:0", shape=(2, 8, 20), dtype=float32)
In [4]: print tf.get_default_graph().get_tensor_by_name("softmax_custom_name:0")
Tensor("softmax_custom_name:0", shape=(16, 20), dtype=float32)
```

Here is the relevant code from the master branch:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/nn_ops.py#L1501
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/nn_ops.py#L1519

Is this the expected behaviour? In this case it would probably make sense to mention it in the documentation?

Otherwise I'd submit a fix.

**Comment by skye** [comment]
It does look like the reshaped output tensor should have "name", rather than its input. Feel free to submit a PR, thanks!
